lib/integration/scenarios/Test/Integration/Scenario/API/Shelley/TransactionsNew.hs

```diff
@@ -3385,6 +3387,96 @@ spec = describe "NEW_SHELLEY_TRANSACTIONS" $ do
                 [ expectField #delegation (`shouldBe` expectedDelegation)
                 ]
 
+    it "TRANS_NEW_JOIN_01f - Cannot re-join the same pool in Babbage"  $ \ctx ->
+        runResourceT $ do
+        noConway ctx "re-joining the same pool outlawed before Conway"
+
+        (src, pool1) <- delegateToPool ctx
+
+        let delegationJoin = Json [json|{
+                "delegations": [{
+                    "join": {
+                        "pool": #{ApiT pool1},
+                        "stake_key_index": "0H"
+                    }
+                }]
+            }|]
+        rTx2 <- request @(ApiConstructTransaction n) ctx
+            (Link.createUnsignedTransaction @'Shelley src) Default delegationJoin
```

Comment:
```suggestion
```

Alas, Daedalus still uses the old `joinStakePool` endpoint

https://cardano-foundation.github.io/cardano-wallet/api/edge/#operation/joinStakePool

Can you write the additional integration tests to be _only_ about this endpoint?

I don't think it makes sense to test `Link.createUnsignedTransaction` at this level — it's a kitchen-sink function, other tests are checking that it's behavior hasn't changed too much; I think that is sufficient.

Reply:
added joinStakePool tests showing the case for both Babbage and Conway era. I would argue testing in new tx workflow is also crucial as we want to be consistent here and outlaw some action (those that are redundant and incur fees)